src/mbgl/renderer/image_manager.cpp

```diff
@@ -73,14 +80,57 @@ void ImageManager::getImages(ImageRequestor& requestor, ImageRequestPair&& pair)
         }
     }
     if (isLoaded() || hasAllDependencies) {
-        notify(requestor, std::move(pair));
+        checkMissingAndNotify(requestor, std::move(pair));
     } else {
         requestors.emplace(&requestor, std::move(pair));
     }
 }
 
 void ImageManager::removeRequestor(ImageRequestor& requestor) {
     requestors.erase(&requestor);
+    missingImageRequestors.erase(&requestor);
+}
+
+void ImageManager::imagesAdded() {
+    for (auto it = missingImageRequestors.begin(); it != missingImageRequestors.end();) {
+        if (it->second.callbacksRemaining == 0) {
+            notify(*it->first, it->second.pair);
+            missingImageRequestors.erase(it++);
+        } else {
+            it++;
```

Comment:
nit: Since the iterator is advanced in both cases, move it to the `for`-clause

Reply:
`missingImageRequestors.erase(it++);` erases the previous iterator after incrementing it. Is rephrasing this to use the for clause possible?

Reply:
You're right, no change is needed here.

Reply:
Yes, this is the correct way to delete the current element while iterating over it.